compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -305,7 +305,6 @@ impl<'tcx> Inliner<'tcx> {
             let func_ty = func.ty(caller_body, self.tcx);
             if let ty::FnDef(def_id, substs) = *func_ty.kind() {
                 // To resolve an instance its substs have to be fully normalized.
-                let substs = self.tcx.try_normalize_erasing_regions(self.param_env, substs).ok()?;
```

Comment:
This normalization is here to avoid the ICE inside of `codegen_select_candidate` called by `resolve_instance`.  The important point is not normalizing, but the `.ok()?`.
If you want to remove the excess normalization, the right way would probably to make `codegen_select_candidate` and `resolve_instance` bubble-up the normalization failure.